docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,205 @@
+
+At the same time computation graphs have been over loaded
+purpose as both a compile-time and run-time data structure.
+The conflation of the description of a computation, the
+representation used for optimizing it, and the data structure
+used to execute it uncessarily hampers many of the goals
+of a framework.
+
+We believe having a high level, an expressive language designed
+for compiler optimizations is essential to the future of an
+end-to-end deep learning compiler stack.
+
+Relay's design was influenced by the authors previous experince
+building advanced optimizing compilers for high level languages,
+as well as experinces with the current TVM stack, and NNVM.
+
+Concretely NNVM has played the role of this high level IR.
+
+for NNVM which address design flaws of the computation graph IRs
+employed by common deep learning frameworks.
+We address a few important challenges. First we present a new IR with first
+class functions and closures, enabling the design of differentiable
+arbitrary control flow constructs. Second we contribute a type system for
+Relay which makes the important properties of computations involving tensors
+explicit.
+
+The result is a high-level IR for differentiable comptuation that can
+be used to abstract over the details of execution including devices,
+parallelism, and distribution.
+
+We believe Relay will enable researchers and industry to design a
+new class of DL framework frontends, differentiable programming languages,
+deep probablistic programming languages, thus accelerating the pace of
+research in applications.
+
+==================
+Language
+==================
+
+Relay is a purely functional, differentiable intermediate representation.
+
+==================
+IR Reference
+==================
+
+The IR has a global environment which stores the set of definitions,
+constants, options, attributes, and provides access to features like
+the type inferencer, constant evaluator, and more.
+
+## Node
+
+The fundmental unit of the IR is the node, all nodes must have
+a type field.
+
+.. code-block:: python
+    class Node:
+        def checked_type() -> Type:
+            ...
+
+==================
+Variables
+==================
+
+Relay has three variable references local, global, and operators. Our design draws inspiration
+from LLVM which differentiates between identifier types. This enables writers of
+optimizations to know precisely what an identifier references without needing information
+beyond the type of identifier.
+
+Globals are written with `@`, locals are written with `%`, and operators are written with
+no sigil like LLVM's intrinsics. The distinction between global and local identifiers
+makes certain kinds of transformation easier. For example inlining a global definition
+requires no analysis, you can write a pass that just directly inlines the defintions.
+It also ensures there are no spooky action at a distance, introducing a new identifier
```

Comment:
Tense does not match: should be "ensures there are no spooky actions at a distance" or "ensures there is no spooky action at a distance." Additionally, the comma should be a semicolon.